### PR TITLE
PR for nop graph fragment deletion and fix for nop graph entry removal from metadb

### PR DESCRIPTION
**Pull request by limetreeestate**

### CODE REVIEW

File: src/server/JasmineGraphInstanceService.cpp [hunk 1]
```diff
@@ -472,6 +472,28 @@ void *instanceservicesession(void *dummyPt) {
             string result = "1";
             write(connFd, result.c_str(), result.size());
             instance_logger.log("Sent : " + result, "info");
+        } else if (line.compare(JasmineGraphInstanceProtocol::DELETE_GRAPH_FRAGMENT) == 0) {
+            /**
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:476]: Lets replace this multi-line comment with single line comment

File: src/server/JasmineGraphInstanceService.cpp [hunk 2]
```diff
@@ -1060,6 +1082,16 @@ void deleteGraphPartition(std::string graphID, std::string partitionID) {
     instance_logger.log("Graph partition and centralstore files are now deleted", "info");
 }
 
+void removeGraphFragments(std::string graphID) {
+    /**
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:1086]: Please move this multi-line comment above the function signature. Please change the multi-line comment following these [guidelines](https://developer.lsst.io/v/DM-5063/docs/cpp_docs.html#function-method-definitions).

File: src/server/JasmineGraphServer.cpp [hunk 3]
```diff
@@ -471,9 +470,33 @@ void JasmineGraphServer::resolveOperationalGraphs(){
 
 }
 
-//TODO:Needs to implement this method
 void JasmineGraphServer::deleteNonOperationalGraphFragment(int graphID) {
+    /**
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:474]: Move this multi-line comment above function signature. Follow the [guideline](https://developer.lsst.io/v/DM-5063/docs/cpp_docs.html#function-method-definitions).

File: src/server/JasmineGraphServer.cpp [hunk 4]
```diff
@@ -471,9 +470,33 @@ void JasmineGraphServer::resolveOperationalGraphs(){
...
+        int port = (it -> second).first;
+        //Initialize threads for host
+        //Each thread runs the service to remove the given graph ID fragments in their datafolders
+        deleteThreads[count++] = std::thread(removeFragmentThroughService, hostname, port, to_string(graphID), this -> masterHost);
+        sleep(1);
+        server_logger.log("Deleted graph fragments of graph ID " + to_string(graphID), "info");
+    }
 
+    for (int threadCount = 0; threadCount < count; threadCount++) {
+        deleteThreads[threadCount].join();
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:497]: Here we should check whether the thread is joinable or not. See this [link](https://github.com/miyurud/jasminegraph/blob/master/src/server/JasmineGraphServer.cpp#L594).

File: src/server/JasmineGraphServer.cpp [hunk 5]
```diff
@@ -1510,6 +1533,99 @@ void JasmineGraphServer::removeGraph(vector<pair<string, string>> hostHasPartiti
     }
 }
 
+int JasmineGraphServer::removeFragmentThroughService(string host, int port, string graphID, string masterIP) {
+    /**
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:1537]: Change the function documentation following this [URL](https://developer.lsst.io/v/DM-5063/docs/cpp_docs.html#function-method-definitions).

File: src/server/JasmineGraphServer.cpp [hunk 6]
```diff
@@ -1510,6 +1533,99 @@ void JasmineGraphServer::removeGraph(vector<pair<string, string>> hostHasPartiti
...
+
+
+        write(sockfd, JasmineGraphInstanceProtocol::DELETE_GRAPH_FRAGMENT.c_str(),
+              JasmineGraphInstanceProtocol::DELETE_GRAPH_FRAGMENT.size());
+        server_logger.log("Sent : " + JasmineGraphInstanceProtocol::DELETE_GRAPH_FRAGMENT, "info");
+        bzero(data, 301);
+        read(sockfd, data, 300);
+        response = (data);
+        response = utils.trim_copy(response, " \f\n\r\t\v");
+        //std::cout << response << std::endl;
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:1607]: Remove the commented line

File: src/server/JasmineGraphInstanceService.cpp (hunk 1 above)
Comment (limetreeestate) [reply src/server/JasmineGraphInstanceService.cpp:476]: Replaced the multi-line comment comment with a single line comment under the else if condition line

File: src/server/JasmineGraphInstanceService.cpp (hunk 2 above)
Comment (limetreeestate) [reply src/server/JasmineGraphInstanceService.cpp:1086]: Followed the guidelines and moved the multi-line comment above the function signature. Used @param to before describing each input parameter. Removed "INPUT" and "OUTPUT" parts of the comment

File: src/server/JasmineGraphServer.cpp (hunk 3 above)
Comment (limetreeestate) [reply src/server/JasmineGraphServer.cpp:474]: Followed the guidelines and moved the multi-line comment above the function signature. Used @param to before describing each input parameter. Removed "INPUT" and "OUTPUT" parts of the comment

File: src/server/JasmineGraphServer.cpp (hunk 4 above)
Comment (limetreeestate) [reply src/server/JasmineGraphServer.cpp:497]: Similar to the line mentioned in the link, checked `deleteThreads[threadCount].joinable()` in an if block before main thread joins.

File: src/server/JasmineGraphServer.cpp (hunk 5 above)
Comment (limetreeestate) [reply src/server/JasmineGraphServer.cpp:1537]: Followed the guidelines and moved the multi-line comment above the function signature. Used `@param` to before describing each input parameter. Removed "INPUT" and "OUTPUT" parts of the function documentation comment

File: src/server/JasmineGraphServer.cpp (hunk 6 above)
Comment (limetreeestate) [reply src/server/JasmineGraphServer.cpp:1607]: Comment removed

File: src/server/JasmineGraphServer.cpp (hunk 5 above)
Comment (miyurud) [reply src/server/JasmineGraphServer.cpp]: Ok

File: src/server/JasmineGraphServer.cpp (hunk 6 above)
Comment (miyurud) [reply src/server/JasmineGraphServer.cpp]: Ok

**Pull request merged**